src/plugins/schema.js

```diff
@@ -0,0 +1,283 @@
+    const { nodes } = block
+    const invalids = nodes.filter(n => n.kind != 'block' && n.kind != 'inline' && n.kind != 'text')
+    return invalids.size ? invalids : null
+  },
+  normalize: (transform, block, invalids) => {
+    return invalids.reduce((t, n) => t.removeNodeByKey(n.key), transform)
+  }
+}
+
+/**
+ * A default schema rule to have at least one text node in blocks/inlines
+ *
+ * @type {Object}
+ */
+
+const MIN_TEXT_RULE = {
+    match: (object) => {
+      return object.kind == 'block' || object.kind == 'inline'
+    },
+    validate: (node) => {
+      const { nodes } = node
+      return nodes.size === 0 ? true : null
+    },
+    normalize: (transform, node) => {
+      return transform.insertNodeByKey(node.key, 0, Text.create())
+    }
+}
+
+/**
+ * A default schema rule to only allow inline and text nodes in inlines.
+ *
+ * @type {Object}
+ */
+
+const INLINE_CHILDREN_RULE = {
+  match: (object) => {
+    return object.kind == 'inline'
+  },
+  validate: (inline) => {
+    const { nodes } = inline
+    const invalids = nodes.filter(n => n.kind != 'inline' && n.kind != 'text')
+    return invalids.size ? invalids : null
+},
+  normalize: (transform, inline, invalids) => {
+    return invalids.reduce((t, n) => t.removeNodeByKey(n.key), transform)
+  }
+}
+
+/**
+ * A default schema rule to ensure that inline nodes are not empty
+ *
+ * @type {Object}
+ */
+
+const INLINE_NO_EMPTY = {
+  match: (object) => {
+    return object.kind == 'inline'
+  },
+  validate: (inline) => {
+    return inline.text == ''
+  },
+  normalize: (transform, node) => {
+    return transform.removeNodeByKey(node.key)
+  }
+}
+
+/**
+ * A default schema rule to ensure that void nodes contain a single space of content.
+ *
+ * @type {Object}
+ */
+
+const INLINE_VOID_TEXT_RULE = {
+  match: (object) => {
+    return (object.kind == 'inline' || object.kind == 'block') && object.isVoid
+  },
+  validate: (node) => {
+    return node.text !== ' ' || node.nodes.size !== 1
+  },
+  normalize: (transform, node) => {
+      transform = node.nodes.reduce((t, child) => {
+          return t.removeNodeByKey(child.key)
+      }, transform)
+
+      return transform.insertNodeByKey(node.key, 0, Text.createFromString(' '))
+  }
+}
+
+/**
+ * A default schema rule to ensure that inline void nodes are surrounded with text nodes
+ *
+ * @type {Object}
+ */
+
+const INLINE_VOID_TEXTS_AROUND_RULE = {
+  match: (object) => {
+    return object.kind == 'block'
+  },
+  validate: (block) => {
+    const invalids = block.nodes.reduce((accu, child, index) => {
+      if (child.kind === 'block' || !child.isVoid) {
+        return accu
+      }
+
+      const prevNode = index > 0 ? block.nodes.get(index - 1) : null
+      const nextNode = block.nodes.get(index + 1)
+
+      const prev = (!prevNode || isInlineVoid(prevNode))
+      const next = (!nextNode || isInlineVoid(nextNode.kind))
+
+      if (next || prev) {
+        accu.push({ next, prev, index })
```

Comment:
I think there's a bug here. The index passed should be shifted for every previously inserted text. Have a counter, incremented by 1 each time `next` is true, and each time `prev` is true.

Then in normalization, the insertion index of `prev` is always the passed `index`, but the insertion index of `next` should be `index + 1` if `prev = false` or `index + 2` if `prev = true`